### DISCUSSION TITLE
Provide the option to operate on Firefox's `extensions.webextensions.restrictedDomains` entries.

### DISCUSSION THREAD

**Discussion by roke-julian-lockhart** [gildas-lormeau/SingleFile#1495] [Ideas]
#### My Rationale

Although [`gildas-lormeau/SingleFile/blob/14da84df6dd3191b31f45c6ecb1dc5c721eea090/faq.md#why-cant-i-save-some-pages-like-httpsaddonsmozillaorgaddonsingle-file`](https://github.com/gildas-lormeau/SingleFile/blob/14da84df6dd3191b31f45c6ecb1dc5c721eea090/faq.md#why-cant-i-save-some-pages-like-httpsaddonsmozillaorgaddonsingle-file) states:

> For security purposes, browsers block web extensions on certain domains. This prevents a malicious extension to remove or change bad reviews, for example.

This is bypassable now. As [`superuser.com/questions/1354603/firefox-extensions-restricted-from-running-on-some-sites#comment2926608_1354612`](https://superuser.com/questions/1354603/firefox-extensions-restricted-from-running-on-some-sites#comment2926608_1354612) explains, [`support.mozilla.org/en-US/kb/quarantined-domains`](https://support.mozilla.org/en-US/kb/quarantined-domains?as=u&utm_source=inproduct#:~:text=how%20can%20i%20re-enable%20the%20add-ons%20that%20are%20not%20allowed%20on%20sites%20restricted%20by%20mozilla%3F) demonstrates that this restriction can be disabled by the user via `about:addons` for add-ons which opt-in.

#### My Desired Solution

Provide this preference.

**Comment by gildas-lormeau** [comment]
I agree this is possible in Firefox. In fact this problem also exists in Chrome and (probably) Safari and I guess I didn't want to indicate every procedure for every browser because it's a problem that's supposed to be rare, and for advanced users. It's also quite easy to get around by using another browser.

**Comment by roke-julian-lockhart** [comment]
> [`#issuecomment-2215255420`](https://github.com/gildas-lormeau/SingleFile/issues/1494#issuecomment-2215255420)

@gildas-lormeau, it's indeed easy to bypass for the content of `extensions.webextensions.restrictedDomains`:

> `accounts-static.cdn.mozilla.net,accounts.firefox.com,addons.cdn.mozilla.net,addons.mozilla.org,api.accounts.firefox.com,content.cdn.mozilla.net,discovery.addons.mozilla.org,install.mozilla.org,oauth.accounts.firefox.com,profile.accounts.firefox.com,support.mozilla.org,sync.services.mozilla.com`

...but not `about:` schema-prefixed URIs, like `about:addons`.

**Comment by gildas-lormeau** [comment]
I'm moving this issue in the Discussions tab because it's not possible from a technical point of view to implement this option in the extension itself.

**Reply by roke-julian-lockhart** [reply]
> [`#discussioncomment-10014748`](https://github.com/gildas-lormeau/SingleFile/discussions/1495#discussioncomment-10014748)

@gildas-lormeau, in retrospect, I believe I actually meant `extensions.quarantinedDomains.enabled`. [^1]

[^1]: [`support.mozilla.org/kb/quarantined-domains`](https://support.mozilla.org/en-US/kb/quarantined-domains)
